src/main/java/org/opentripplanner/routing/algorithm/astar/AStar.java

```diff
@@ -69,7 +71,7 @@ protected AStar(
     this.timeout = timeout;
 
     this.spt = new ShortestPathTree(dominanceFunction);
-    this.heuristic.initialize(request, fromVertices, toVertices);
+    this.heuristic.initialize(request, streetMode, fromVertices, toVertices);
```

Comment:
This can be done in the builder, then there is no need to pass in the RouteRequest into AStar any more, just the arriveBy flag.

Reply:
done